core/arch/arm/kernel/thread_a64.S

```diff
@@ -986,6 +987,16 @@ END_FUNC el0_sync_abort
 	msr	spsel, #0
 	mov	sp, x1
 
+#ifdef CFG_CORE_WORKAROUND_NSITR_CACHE_PRIME
+	/*
+	 * Prevent leaking information about which entries has been used in
+	 * cache. We're relying on the dispatcher in TF-A to take care of
+	 * the BTB.
+	 */
+	mov	x0, #DCACHE_OP_CLEAN_INV
```

Comment:
Will only Invalidating (not clean to louis) only secure memory (not the entire L1 cache) range do the job ?

Reply:
We must clean or we'll lose data. At this stage it's a bit hard to predict which virtual memory ranges has been in use so we'd need to do all possible ranges which is many MiBs of virtual memory.

Reply:
yes, clean cannot be avoided.
Also if tzdram size exceeds L1D/I cache size then range flush can be more expensive.

Reply:
But I guess primed L2 is still vulnerable to such attack unless range flush is done.

Reply:
> But I guess primed L2 is still vulnerable to such attack unless range flush is done.

I was wondering the same. And in fact, the "Future Work" sections mentions it: "It may be possible to infer more detailed information about victim execution by targeting other shared microarchitectural structures, such as the [...] last-level cache (LLC)."

Reply:
Yes, but I'm not sure how effective it is.